ghc/Main.hs

```diff
@@ -522,59 +536,68 @@ type ModeM = CmdLineP (Maybe (Mode, String), [String], [Located String])
+               , Flag "V"                     (PassFlag (setMode showVersionMode))
+               , Flag "-version"              (PassFlag (setMode showVersionMode))
+               , Flag "-numeric-version"      (PassFlag (setMode showNumVersionMode))
+               , Flag "-info"                 (PassFlag (setMode showInfoMode))
+               , Flag "-show-options"         (PassFlag (setMode showOptionsMode))
+               , Flag "-supported-languages"  (PassFlag (setMode showSupportedExtensionsMode))
+               , Flag "-supported-extensions" (PassFlag (setMode showSupportedExtensionsMode))
+               ]
+        othr = [ Flag k'                      (PassFlag (setMode (printSetting k)))
+               | k <- ["Project version",
+                       "Booter version",
+                       "Stage",
+                       "Build platform",
+                       "Host platform",
+                       "Target platform",
+                       "Have interpreter",
+                       "Object splitting supported",
+                       "Have native code generator",
+                       "Support SMP",
+                       "Unregisterised",
+                       "Tables next to code",
+                       "RTS ways",
+                       "Leading underscore",
+                       "Debug on",
+                       "LibDir",
+                       "Global Package DB",
+                       "C compiler flags",
+                       "Gcc Linker flags",
+                       "Ld Linker flags"],
+                 let k' = "-print-" ++ map (replaceSpace . toLower) k
+                     replaceSpace ' ' = '-'
+                     replaceSpace c   = c
+               ]
+        ------- interfaces -----------------------------------------------------
+        prim = [ Flag "-show-iface"  (HasArg (\f -> setMode (showInterfaceMode f)
+                                                            "--show-iface"))
+
+        ------- primary modes --------------------------------------------------
+               , Flag "c"            (PassFlag (\f -> do setMode (stopBeforeMode StopLn) f
+                                                         addFlag "-no-link" f))
+               , Flag "M"            (PassFlag (setMode doMkDependHSMode))
+               , Flag "E"            (PassFlag (setMode (stopBeforeMode anyHsc)))
+               , Flag "C"            (PassFlag (setMode (stopBeforeMode HCc)))
+               , Flag "S"            (PassFlag (setMode (stopBeforeMode (as False))))
+               , Flag "-make"        (PassFlag (setMode doMakeMode))
+               , Flag "-interactive" (PassFlag (setMode doInteractiveMode))
+               , Flag "-abi-hash"    (PassFlag (setMode doAbiHashMode))
+               , Flag "e"            (SepArg   (\s -> setMode (doEvalMode s) "-e"))
+               ]
+#if __GLASGOW_HASKELL__ >= 709
+        ghcModes = cycle [AllModes]
```

Comment:
@chrisdone - I just guessed the `AllModes` flag.
https://downloads.haskell.org/~ghc/latest/docs/html/libraries/ghc-7.10.1/CmdLineParser.html#v:AllModes